x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -283,6 +284,17 @@ private void preAnalyzeIndices(
             var fieldNames = fieldNames(parsed, enrichPolicyMatchFields);
 
             Map<String, OriginalIndices> clusterIndices = indicesExpressionGrouper.groupIndices(IndicesOptions.DEFAULT, table.index());
+            // If groupIndices() could not resolve the wildcard-ed cluster, it returns the local cluster
+            // and sets the indices to an empty array. If this scenario is not explicitly handled, any
+            // searches involving the unresolved cluster will fall back and return results from local
+            // cluster.
+            if (clusterIndices.size() == 1) {
+                OriginalIndices localCluster = clusterIndices.get(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+                if (localCluster != null && localCluster.indices() == Strings.EMPTY_ARRAY) {
+                    throw new VerificationException("No matching clusters " + table.index());
```

Comment:
Can you put the indexExpression in brackets? That's a typical error message practice in Elasticsearch. So it should be:
```
No matching clusters [xremote*:web_traffic]
```